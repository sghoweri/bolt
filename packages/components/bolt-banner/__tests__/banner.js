import {
  isConnected,
  render,
  renderString,
  stopServer,
  html,
} from '../../../../packages-extras/testing/testing-helpers';
import schema from '../banner.schema';
const { status, align } = schema.properties;
const timeout = 120000;

describe('<bolt-banner> Component', () => {
  afterAll(async () => {
    await stopServer();
  }, 100);

  // Basic Usage
  test('Basic usage', async () => {
    const results = await render('@bolt-components-banner/banner.twig', {
      content: 'This is a banner',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  // Props
  status.enum.forEach(async statusChoice => {
    test(`Status of the banner message: ${statusChoice}`, async () => {
      const results = await render('@bolt-components-banner/banner.twig', {
        content: `This banner is trying to convey ${statusChoice}`,
        status: statusChoice,
      });
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });

  align.enum.forEach(async alignChoice => {
    test(`Text alignment: ${alignChoice}`, async () => {
      const results = await render('@bolt-components-banner/banner.twig', {
        content: `The text is aligned to the ${alignChoice}`,
        align: alignChoice,
      });
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });
});
