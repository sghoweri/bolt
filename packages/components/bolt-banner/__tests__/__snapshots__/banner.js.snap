// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bolt-banner> Component Basic usage 1`] = `
<bolt-banner>
  <div class="c-bolt-banner c-bolt-banner--status-information   c-bolt-banner--align-center c-bolt-banner--full">
    <ssr-keep for="bolt-banner">
      This is a banner
    </ssr-keep>
  </div>
</bolt-banner>
`;

exports[`<bolt-banner> Component Status of the banner message: error 1`] = `
<bolt-banner status="error">
  <div class="c-bolt-banner c-bolt-banner--status-error t-bolt-dark  c-bolt-banner--align-center c-bolt-banner--full">
    <ssr-keep for="bolt-banner">
      This banner is trying to convey error
    </ssr-keep>
  </div>
</bolt-banner>
`;

exports[`<bolt-banner> Component Status of the banner message: information 1`] = `
<bolt-banner status="information">
  <div class="c-bolt-banner c-bolt-banner--status-information   c-bolt-banner--align-center c-bolt-banner--full">
    <ssr-keep for="bolt-banner">
      This banner is trying to convey information
    </ssr-keep>
  </div>
</bolt-banner>
`;

exports[`<bolt-banner> Component Status of the banner message: success 1`] = `
<bolt-banner status="success">
  <div class="c-bolt-banner c-bolt-banner--status-success  t-bolt-dark c-bolt-banner--align-center c-bolt-banner--full">
    <ssr-keep for="bolt-banner">
      This banner is trying to convey success
    </ssr-keep>
  </div>
</bolt-banner>
`;

exports[`<bolt-banner> Component Status of the banner message: warning 1`] = `
<bolt-banner status="warning">
  <div class="c-bolt-banner c-bolt-banner--status-warning   c-bolt-banner--align-center c-bolt-banner--full">
    <ssr-keep for="bolt-banner">
      This banner is trying to convey warning
    </ssr-keep>
  </div>
</bolt-banner>
`;

exports[`<bolt-banner> Component Text alignment: center 1`] = `
<bolt-banner align="center">
  <div class="c-bolt-banner c-bolt-banner--status-information   c-bolt-banner--align-center c-bolt-banner--full">
    <ssr-keep for="bolt-banner">
      The text is aligned to the center
    </ssr-keep>
  </div>
</bolt-banner>
`;

exports[`<bolt-banner> Component Text alignment: end 1`] = `
<bolt-banner align="end">
  <div class="c-bolt-banner c-bolt-banner--status-information   c-bolt-banner--align-end c-bolt-banner--full">
    <ssr-keep for="bolt-banner">
      The text is aligned to the end
    </ssr-keep>
  </div>
</bolt-banner>
`;

exports[`<bolt-banner> Component Text alignment: start 1`] = `
<bolt-banner align="start">
  <div class="c-bolt-banner c-bolt-banner--status-information   c-bolt-banner--align-start c-bolt-banner--full">
    <ssr-keep for="bolt-banner">
      The text is aligned to the start
    </ssr-keep>
  </div>
</bolt-banner>
`;
